lib/utils.js

```diff
@@ -134,40 +134,91 @@ module.exports = {
   },
 
   /**
+   *
+   * References:
+   * - https://github.com/facebook/react-native/blob/main/Libraries/Utilities/Platform.ios.js
+   * - https://github.com/facebook/react-native/blob/main/Libraries/Utilities/NativePlatformConstantsAndroid.js
+   *
    * @returns An object with names and versions of the various components making up the context.
    */
   getVersions() {
     const packageJson = require("../package.json");
-    const packageVersion = packageJson.version;
+    const sdkVersion = packageJson.version;
     const environment = this.getEnvironment();
 
+    const sdk = "JS";
+
     try {
       if (environment === "reactnative") {
         const { Platform } = require("react-native");
+        let deviceName = "unknown";
+        let deviceVersion = "unknown";
+        if (Platform.OS === "ios") {
+          if (Platform.isPad()) {
+            deviceName = "iPad";
+          } else if (Platform.isTV()) {
+            deviceName = "AppleTV";
+          } else {
+            // RN doesn't support Apple Watch
+            deviceName = "iPhone";
+          }
+          deviceVersion = "unknown";
+        } else if (Platform.OS === "android") {
+          deviceName = `${Platform.Manufacturer}:${Platform.Brand}`;
+          deviceVersion = `${Platform.Model}`;
+        }
+
         return {
-          packageVersion,
-          platformContext: environment,
-          platformOs: Platform.OS,
+          sdk,
+          sdkVersion,
+
+          platform: Platform.OS,
           // Android reports a number ...
           platformVersion: `${Platform.Version}`,
+
+          deviceName,
+          deviceVersion,
+
+          cpuArch: "unknown", // Detecting CPU architecture on Android and iOS is done by seperate methods in platform.{mm, cpp)
+
+          frameworkName: "RN",
```

Comment:
Is this required by some spec? I think it would be most useful for a consumer of these metrics to not have it abbreviated.

```suggestion
          frameworkName: "React Native",
```

```suggestion
          frameworkName: "react-native",
```